secure-staff-services.md

```diff
@@ -0,0 +1,65 @@
+# Securing staff facing services
+
+## Context 
+
+We have a need to make services securely available to both our editorial and engineering staff. In the past we have relied on IP restricting access to internal IP ranges accessible on premises and via VPN by [AWS Security Groups](https://docs.aws.amazon.com/managedservices/latest/userguide/about-security-groups.html).
+
+Restricting access to IP ranges as a method of protecting our tools became impractical after the [network incident at the start of 2023](https://www.theguardian.com/media/2022/dec/21/guardian-hit-by-serious-it-incident-believed-to-be-ransomware-attack). 
+
+In addition the security offered by IP restriction relies on trusting all access from within the network which is less secure than authenticating each users identity with a trusted identity provider.
+
+In order to improve the security of our services, and restore access to previously IP restricted services in the absence of a working VPN we have established this pattern in conjunction for InfoSec.
+
+## Problem statement
+
+To improve security for our staff tools available on the internet we need to be able to: 
+
+- Notice and react to potential attacks that come via web requests
+- Log requests to all our services so that they can be centrally audited
+- Authenticate external requests before they reach our applications
+- Authenticate requests that come from within our network
+- Handle authorization and access management to restrict access to specific users
+
+## Solution
+
+This solution brings together multiple components to provide a single secure solution:
+
+### Authentication at the ALB
+
+For the case where an application is served via an [Application Load Balancer (ALB)](https://docs.aws.amazon.com/elasticloadbalancing/latest/application/introduction.html) in AWS we will use [authentication at the ALB](https://docs.aws.amazon.com/elasticloadbalancing/latest/application/listener-authenticate-users.html) with Google as an OpenID Connect (OIDC) compliant Identity Provider (IdP). This will require a Google OAuth client to be configured for your application as part of a Google Cloud project. 
+
+This approach allows us to authenticate external requests before they reach our applications.
+
+A proposed extension to this would be this [Cognito authentication proposal](https://docs.google.com/document/d/1bEyL_Hn7DRs7XNyhvjvDK0PGXRzxgFzCbMcnaVyXbA0/edit#heading=h.nchxv8vlw2w5) that provides further custom authentication with Google as an IdP but allows potential access management based on Google groups, and enforces MFA requirements. 
```

Comment:
I think there should be some logic behind the sensitivity of the service and whether locking it down to just the organisation is sufficient, or if we should require membership of a group and 2fa enabled. I think it would be good to mention what qualifies a service for a level of protection, i.e. it's a read only service and doesn't expose particularly sensitive data (Amiable) or is it a service which can make changes (riff-raff) and should therefore be restricted by group.